enumerable_methods.rb

```diff
@@ -0,0 +1,174 @@
+module Enumerable
+  # start of my_each
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    until i >= length
+      yield self[i]
+      i += 1
+    end
+    self
+  end
+  # end of my_each
+
+  # start of my_each_with_index
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    until i >= length
+      yield self[i], i
+      i += 1
+    end
+    self
+  end
+  # end of my_each_with_index
+
+  # my_select
+  def my_select
+    return to_enum unless block_given?
+
+    new_array = []
+    i = 0
+
+    until i >= length
+      new_array << self[i] if yield self[i]
+      i += 1
+    end
+    new_array
+  end
+  # end of my_select
+
+  # my_all?
+  def my_all?
+    my_each do |item|
+      return false unless yield item
+    end
+    true
+  end
+  # end of my_all?
+
+  # my_any?
+  def my_any?
+    my_each do |item|
+      return true if yield item
+    end
+    false
+  end
+  # end of my_any?
+
+  # my_none?
+  def my_none?
+    my_each do |item|
+      return false if yield item
+    end
+    true
+  end
+  # end of my_none?
+
+  # my_count
+  def my_count(arg = nil)
+    count = 0
+
+    if block_given?
+      my_each do |ele|
+        count += 1 if yield(ele) == true
+      end
+
+    elsif arg.nil?
+      my_each do |_ele|
+        count += 1
+      end
+
+    else
+      my_each do |ele|
+        count += 1 if ele == arg
+      end
+    end
+    count
+  end
+  # end of my_count
+
+  # start of my_map
+
+  def my_map(_proc = nil)
```

Comment:
[OPTIONAL]

- [ ] In the specifications of the project in the [Odin project](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks) # 12, you were asked to modify your my_map method to take either a proc or a block. And if both a proc and a block are given, only execute the proc. The real map doesn't have this feature, but you were asked to have it so I strongly recommend working on it.
Given 
```ruby
my_proc = Proc.new {|num| num > 10 }
```
Result
```ruby
puts [18, 22, 5, 6] .my_map(my_proc) {|num| num < 10 }  => true true false false
```